tracing-opentelemetry/src/span_ext.rs

```diff
@@ -86,6 +120,26 @@ impl OpenTelemetrySpanExt for tracing::Span {
         });
     }
 
+    fn add_link(&self, cx: Context) {
+        let mut cx = Some(cx);
+        self.with_collector(move |(id, collector)| {
+            if let Some(get_context) = collector.downcast_ref::<WithContext>() {
+                get_context.with_context(collector, id, move |builder, _tracer| {
+                    if let Some(cx) = cx.take() {
+                        let follows_context = cx.span().span_context().clone();
+                        let follows_link =
+                            opentelemetry::trace::Link::new(follows_context, Vec::new());
```

Comment:
I notice this will always create a link with no key-value attributes. Do you think it makes sense to include a separate function like `add_link_with_attributes(&self, cx: Context, attrs: Vec<KeyValue>)`, or something?

We can always add this in a separate PR, of course.

Reply:
`add_link_with_attributes` sounds good to me 👍